packages/examples/main/src/python/client.ts

```diff
@@ -0,0 +1,153 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2018-2022 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import 'monaco-editor/esm/vs/editor/editor.all.js';
+import 'monaco-editor/esm/vs/editor/standalone/browser/iPadShowKeyboard/iPadShowKeyboard.js';
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api.js';
+import * as vscode from 'vscode';
+import 'vscode/default-extensions/theme-defaults';
+import 'vscode/default-extensions/python';
+import { updateUserConfiguration } from 'vscode/service-override/configuration';
+import { createConfiguredEditor, createModelReference } from 'vscode/monaco';
+import { ExtensionHostKind, registerExtension } from 'vscode/extensions';
+import { initServices, MonacoLanguageClient } from 'monaco-languageclient';
+import { CloseAction, ErrorAction, MessageTransports } from 'vscode-languageclient';
+import { WebSocketMessageReader, WebSocketMessageWriter, toSocket } from 'vscode-ws-jsonrpc';
+import { RegisteredFileSystemProvider, registerFileSystemOverlay, RegisteredMemoryFile } from 'vscode/service-override/files';
+
+import { buildWorkerDefinition } from 'monaco-editor-workers';
+buildWorkerDefinition('../../../node_modules/monaco-editor-workers/dist/workers/', new URL('', window.location.href).href, false);
+
+const languageId = 'python';
+let languageClient: MonacoLanguageClient;
+
+const createWebSocket = (url: string): WebSocket => {
+    const webSocket = new WebSocket(url);
+    webSocket.onopen = async () => {
+        const socket = toSocket(webSocket);
+        const reader = new WebSocketMessageReader(socket);
+        const writer = new WebSocketMessageWriter(socket);
+        languageClient = createLanguageClient({
+            reader,
+            writer
+        });
+        await languageClient.start();
+        reader.onClose(() => languageClient.stop());
+
+        vscode.commands.registerCommand('pyright.restartserver', (...args: unknown[]) => {
+            languageClient.sendRequest('workspace/executeCommand', { command: 'pyright.restartserver', arguments: args });
+        });
```

Comment:
What if the socket is recreated? is the command overriden?

Maybe storing the last languageClient in a variable and registering the command at the root is more common

Reply:
True, I moved the registration to the run function and extracted the function. This way it always uses the current languageclient.